src/app/utils/dataLoaderUtil.js

```diff
@@ -0,0 +1,141 @@
+import Actions from '@Actions';
+import { ajaxCall, destructureFilters } from '@utils';
+import { pick as _pick } from 'underscore';
+import appConfig from '@appConfig';
+import Store from '@Store';
+
+const pathInstructions = [
+  {
+    expression: /\/research\/collections\/shared-collection-catalog\/bib\/([cp]?b\d*)/,
+    pathType: 'bib',
+  },
+  {
+    expression: /\/research\/collections\/shared-collection-catalog\/search\?(.*)/,
+    pathType: 'search',
+  },
+  {
+    expression: /\/research\/collections\/shared-collection-catalog\/hold\/request\/([^/]*)/,
+    pathType: 'holdRequest',
+  },
+];
+
+const routePaths = {
+  bib: `${appConfig.baseUrl}/api/bib`,
+  search: `${appConfig.baseUrl}/api`,
+  holdRequest: `${appConfig.baseUrl}/api/hold/request/:bibId-:itemId`,
+};
+
+const routesGenerator = location => ({
+  bib: {
+    apiRoute: (matchData, route) => `${route}?bibId=${matchData[1]}`,
+    actions: [Actions.updateBib],
+    errorMessage: 'Error attempting to make an ajax request to fetch a bib record from ResultsList',
+  },
+  search: {
+    apiRoute: (matchData, route) => `${route}?${matchData[1]}`,
+    actions: [
+      data => Actions.updateSearchResults(data.searchResults),
+      () => Actions.updatePage(location.query.page || 1),
+      () => Actions.updateSearchKeywords(location.query.q),
+      data => Actions.updateFilters(data.filters),
+      (data) => {
+        if (data.filters && data.searchResults) {
+          const unescapedQuery = Object.assign(
+            {},
+            ...Object.keys(location.query)
+              .map(k => ({ [decodeURIComponent(k)]: decodeURIComponent(location.query[k]) })),
+          );
+          const urlFilters = _pick(unescapedQuery, (value, key) => {
+            if (key.indexOf('filter') !== -1) {
+              return value;
+            }
+            return null;
+          });
+          Actions.updateSelectedFilters(destructureFilters(urlFilters, data.filters));
```

Comment:
`destructureFilters` is buggy and has not actually been used in the code. In most places `data.filters` is passed directly to `Actions.updateSelectedFilters`